src/backend/distributed/connection/connection_management.c

```diff
@@ -473,50 +475,37 @@ FindAvailableConnection(dlist_head *connections, uint32 flags)
 		{
 			/*
 			 * The caller requested a metadata connection, and this is not the
+			 * metadata connection. Still, this is a candidate for becoming a
 			 * metadata connection.
 			 */
+			metadataConnectionCandidateList =
+				lappend(metadataConnectionCandidateList, connection);
 			continue;
 		}
-		else
```

Comment:
else part is removed mostly for ease of readability. We do `AssertSingleMetadataConnectionExists` below right after `metadataConnection->useForMetadataOperations = true;`. 

So, this part seemed not very useful